### PR TITLE
Update `Auto-sizes for Lazy-loaded Images` plugin name to `Enhanced Responsive Images`

### LINKED ISSUE #1330: Update plugin name for Initial work on Accurate Sizes in Auto Sizes Plugin
As we are in the phase where we are about to merge the initial work for accurate sizes for images in the Auto Sizes plugin, we should update the plugin name to reflect these changes so the end user knows that initial changes have been implemented. Additionally, we should add an Upgrade Notice that references these changes.

https://github.com/WordPress/performance/pull/1329

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
## Summary

Fixes #1330

### CODE REVIEW

**Review by adamsilverstein — APPROVED** [review]
Nice!

**Review by thelovekesh — APPROVED** [review]

File: plugins/auto-sizes/tests/test-auto-sizes.php [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?php
 /**
- * Tests for the Auto-sizes for Lazy-loaded Images plugin.
+ * Tests for the Auto-sizes for Lazy-loaded Images.
```
Comment (westonruter) [inline plugins/auto-sizes/tests/test-auto-sizes.php]:
```suggestion
 * Tests for the Enhanced Responsive Images plugin.
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -26,8 +26,8 @@
 /tests/testdata/modules/site-health/audit-enqueued-assets                   @manuelRod
 
 # Plugin: Auto-sizes for Lazy-Loaded Images
```
Comment (westonruter) [inline .github/CODEOWNERS]:
```suggestion
# Plugin: Enhanced Responsive Images
```

**Review by westonruter — APPROVED** [review]

**Comment by joemcgill** [comment]
Going to merge this change into the feature branch while we do final prep.

**Pull request merged**